server/models/listing.model.js

```diff
@@ -0,0 +1,50 @@
+const { PrismaClient } = require('@prisma/client')
+
+const prisma = new PrismaClient({log: ['query', 'info', 'warn', 'error'],  errorFormat: "minimal",}
+)
+
+async function newListing(id, petName){
+         //Check required properties
+    if (!id || !petName ){
+        return ({
+           error: 'Missing required properties.'
+        })
+    }
+    //Check if username doesn't exist already
+   //This check is done by enforcing a unique constraint on username(email) in the DB, done at the time of user creation
+
+        try {
+            const newListing = await  prisma.listing.create({
+                data: {
```

Comment:
I am assuming this is for testing purposes ? Because the params come from the controller (req.body.petName, req.body.petPhoto etc)

Reply:
Yes, it's a test. We're gonna continue working on that this afternoon with Carlos. Thank you for the useful insights!